assets/scripts/streets/StreetName.jsx

```diff
@@ -44,10 +45,61 @@ class StreetName extends React.PureComponent {
     super(props)
 
     this.state = {
-      isHovered: false
+      isHovered: false,
+      modify: false,
+      inpVal: ''
     }
   }
 
+  handleStreetNameChange = () => {
+    let newName = this.state.inpVal
+    if (newName) {
+      this.props.handleStreetNameChange(newName)
+      this.setState({
+        modify: false,
+        inpVal: '',
+        isHovered: false
+      })
+    }
+  }
+
+  handleKeyUp = (evt) => {
+    if (evt.key === 'Enter') {
+      this.handleStreetNameChange()
+    }
+    if (evt.key === 'Escape') {
+      this.setState({
+        modify: false,
+        inpVal: '',
+        isHovered: false
+      })
```

Comment:
This specific `setState()` occurs enough times in this component that a specific `this.resetState()` method may be very useful.